controller/Publish.php

```diff
@@ -61,6 +63,19 @@ public function publish(){
             $deliveryClass = $this->getClass($classUri);
             /** @var DeliveryFactory $deliveryFactoryResources */
             $deliveryFactoryResources = $this->getServiceManager()->get(DeliveryFactory::SERVICE_ID);
+
+            $qtiTestService = $this->getServiceLocator()->get(TestModelService::SERVICE_ID);
+            $items = $qtiTestService->getItems($test);
+            $lang = \common_session_SessionManager::getSession()->getDataLanguage();
+
+            foreach ($items as $item) {
+                $qtiItem = QtiItemService::singleton()->getDataItemByRdfItem($item, $lang, true);
```

Comment:
Please take the `QtiItemService::singleton()` part out of the `foreach` to invoke it only once.